src/ast/ast.h

```diff
@@ -533,6 +537,46 @@ class Variable : public Node {
   SizedType var_type;
 };
 
+class VariableAddr : public Node {
+public:
+  explicit VariableAddr(ASTContext &ctx, Variable *var, Location &&loc)
+      : Node(ctx, std::move(loc)),
+        var(var),
+        var_addr_type(CreatePointer(CreateNone())) {};
+  explicit VariableAddr(ASTContext &ctx,
+                        const VariableAddr &other,
+                        const Location &loc)
+      : Node(ctx, loc + other.loc),
+        var(other.var),
+        var_addr_type(other.var_addr_type) {};
+
+  const SizedType &type() const
+  {
+    assert(var_addr_type.IsPtrTy());
+    return var_addr_type;
+  }
+
+  Variable *var = nullptr;
+  // This should always be a pointer type
```

Comment:
Nit: doesn't feel like a useful comment to me. There are all kinds of constraints on the types imposed during semantic analysis

Reply:
You mean we can't use just `CreatePointer(var->var_type)` in those places?